src/conn/conn_layered.c

```diff
@@ -532,12 +536,23 @@ __disagg_pick_up_checkpoint(WT_SESSION_IMPL *session, uint64_t meta_lsn)
     WT_ERR(__wt_strdup(session, root, &conn->disaggregated_storage.last_checkpoint_root));
 
     /* Keep a record of past checkpoints, they will be needed for ingest garbage collection. */
-    WT_ERR(__layered_track_checkpoint(session, checkpoint_timestamp));
+    WT_ERR_MSG_CHK(session, __layered_track_checkpoint(session, checkpoint_timestamp),
+      "No checkpoint found for track");
 
     /* Update ingest tables' prune timestamps. */
-    WT_ERR(__layered_update_gc_ingest_tables_prune_timestamps(internal_session));
+    WT_ERR_MSG_CHK(session, __layered_update_gc_ingest_tables_prune_timestamps(internal_session),
+      "No checkpoint found for update");
 
 err:
+    /* Add stat for success check here */
```

Comment:
I would remove this comment, the code is self-explanatory. Maybe you added it when you started on the ticket.
```suggestion
```